model/riscv_sys_regs.sail

```diff
@@ -543,6 +543,53 @@ function legalize_mcountinhibit(c : Counterin, v : xlenbits) -> Counterin = {
   c
 }
 
+bitfield CountSmcntrpmf : bits(64) = {
+  MINH  : 62,
+  SINH  : 61,
+  UINH  : 60,
+  VSINH : 59,
+  VUINH : 58
+}
+register mcyclecfg : CountSmcntrpmf
+
+function should_inc_mcycle() -> bool = {
+  match (cur_privilege) {
+    Machine    => mcyclecfg.MINH() != 0b1,
+    Supervisor => mcyclecfg.SINH() != 0b1,
+    User       => mcyclecfg.UINH() != 0b1
+    /* TODO: VSINH, VUINH when those modes are defined */
+  }
+}
+
+register minstretcfg : CountSmcntrpmf
+
+function should_inc_minstret(orig_priv : Privilege) -> bool = {
+  match (orig_priv) {
+    Machine    => minstretcfg.MINH() != 0b1,
+    Supervisor => minstretcfg.SINH() != 0b1,
+    User       => minstretcfg.UINH() != 0b1
+    /* TODO: VSINH, VUINH when those modes are defined */
+  }
+}
+
+function legalize_smcntrpmf(c : CountSmcntrpmf, high: bool, value : xlenbits) -> CountSmcntrpmf = {
+  if high == false & sizeof(xlen) == 32 then
```

Comment:
We probably should give more thought into a nicer way to handle all these *h CSRs being added

Reply:
Now that I have a bit more SAIL experience, I took a stab at doing something better here. Please take a look.